### PR TITLE
Refactor to remove wp eval code

### PR DESCRIPTION
**Pull request by javierdelapuente**
Applicable spec: <link>

### Overview

Small refactor to remove `wp eval` that runs directly code against the wordpress installation to configure the `wordpress-teams-integration` plugin. 

Instead, to configure options  in the same way in all plugins, use `wp option update` command instead of `eval`. That will make it easier to have generic code, and stop using hardcoded php code to configure options in the plugin.

A change was also needed  in the plugin (already merged):
https://git.launchpad.net/~canonical-sysadmins/wordpress-teams-integration/+git/wordpress-teams-integration/commit/?id=a643325df2e40ad5a34346f93872c47cdaa09051

### Checklist

- [x] The [charm style guide](https://juju.is/docs/sdk/styleguide) was applied
- [x] The [contributing guide](https://github.com/canonical/is-charms-contributing-guide) was applied
- [x] The changes are compliant with [ISD054 - Managing Charm Complexity](https://discourse.charmhub.io/t/specification-isd014-managing-charm-complexity/11619)
- [x] The documentation is generated using `src-docs`
- [x] The documentation for charmhub is updated.
- [x] The PR is tagged with appropriate label (`urgent`, `trivial`, `complex`)

### CODE REVIEW

**Review by arturo-seijas — APPROVED** [review]

File: src/charm.py [hunk 1]
```diff
@@ -1199,21 +1188,20 @@ def _encode_openid_team_map(team_map: str) -> str:
...
-                f"'team'=>'{launchpad_role}',"
-                f"'role'=>'{wordpress_role}',"
-                f"'server' => '0',),"
-            )
-        return f"array({''.join(array_items)})"
+            teams_parsed[str(idx)] = {
+                "id": idx,
+                "team": launchpad_role,
+                "role": wordpress_role,
+                "server": "0",
```
Comment (yanksyoon) [inline src/charm.py:1202]: Out of curiosity, what does the server represent? - (Optionally, would it be possible to have this in a typed dict with attributes so these can be explained within docstrings?)

**Review by yanksyoon — APPROVED** [review]
LGTM!

**Review by amandahla — APPROVED** [review]
LGTM

File: src/charm.py (hunk 1 above)
Comment (javierdelapuente) [reply src/charm.py:1202]: Looks like it is a way of mapping servers to [open id endpoints](https://git.launchpad.net/~canonical-sysadmins/wordpress-teams-integration/+git/wordpress-teams-integration/tree/openid-teams.php?id=a643325df2e40ad5a34346f93872c47cdaa09051#n50) and 0 is the only value used... To be honest, it looks like something not used...

I will merge as I do not plan to work more in Wordpress charm, but I believe the correct action, beside documenting as you state, would be to work on the plugins...

**Pull request merged**